src/pages/ProductList/ProductList.js

```diff
@@ -0,0 +1,108 @@
+import React, { useState, useEffect } from "react";
+import Layout from "../../components/layout/Layout";
+import { useProducts } from "../../utils/hooks/useProducts";
+import Products from "../../components/Products/Products";
+import { useSearchParams } from "react-router-dom";
+import Pagination from "../../components/Pagination";
+
+const ProductList = () => {
+  const [productQuery, setProductQuery] = useState({
+    categories: [],
+    pageSize: 12,
+    currentPage: 1,
+  });
+  const { data, isLoading } = useProducts(productQuery);
+  const [searchParams] = useSearchParams();
+  const [checkedItems, setCheckedItems] = useState([]);
+  const [currentPage, setCurrentPage] = useState(1);
+  const searchTerm = searchParams.get("searchTerm");
+  const [isSearcResult, setIsSearcResult] = useState(false);
+  useEffect(() => {
+    const queryCategory = searchParams.get("category");
+    if (queryCategory !== null) {
+      setCheckedItems((prevCheckedItems) => [queryCategory]);
+      setProductQuery((prevProductQuery) => ({
+        ...prevProductQuery,
+        categories: [queryCategory],
+      }));
+    }
+  }, [searchParams]);
+
+  useEffect(() => {
+    setProductQuery((prevProductQuery) => ({
+      ...prevProductQuery,
+      currentPage,
+    }));
+  }, [currentPage]);
+
+  useEffect(() => {
+    if (searchTerm == null || searchTerm === "") {
+      setProductQuery((prevProductQuery) => ({
+        ...prevProductQuery,
+        searchTerm: null,
+        pageSize: 12,
+      }));
+      setIsSearcResult((prevIsSearcResult) => false);
+    } else {
+      setProductQuery((prevProductQuery) => ({
+        ...prevProductQuery,
+        searchTerm,
+        pageSize: 20,
+      }));
+      setIsSearcResult((prevIsSearcResult) => true);
+    }
```

Comment:
this `if/else` is not required, with the ternary you can change the params passed to update the state:

```suggestion
    setProductQuery((prevProductQuery) => ({
      ...prevProductQuery,
      searchTerm: searchTerm || null,
      pageSize: searchTerm ? 20 : 12,
    }));
    setIsSearcResult(!!searchTerm);
```